Source/Core/VideoCommon/Present.cpp

```diff
@@ -23,13 +23,74 @@
 
 std::unique_ptr<VideoCommon::Presenter> g_presenter;
 
+// The video encoder needs the image to be a multiple of x samples.
+static constexpr int VIDEO_ENCODER_LMC = 4;
+
 namespace VideoCommon
 {
 static float AspectToWidescreen(float aspect)
 {
   return aspect * ((16.0f / 9.0f) / (4.0f / 3.0f));
 }
 
+static std::tuple<int, int> FindClosestIntegerResolution(float width, float height,
+                                                         float aspect_ratio)
+{
+  // We can't round both the x and y resolution as that might generate an aspect ratio
+  // further away from the target one, we also can't either ceil or floor both sides,
+  // so we find the combination or flooring and ceiling that is closest to the target ar.
+  const int ceiled_width = static_cast<int>(std::ceil(width));
+  const int ceiled_height = static_cast<int>(std::ceil(height));
+  const int floored_width = static_cast<int>(std::floor(width));
+  const int floored_height = static_cast<int>(std::floor(height));
+
+  const float ceil_ceil_aspect_ratio = static_cast<float>(ceiled_width) / ceiled_height;
```

Comment:
What you have is fine.  But I wonder if this would work and maybe be a bit simpler?

```
int int_width;
int int_height;
float min_distance = std::numeric_limits<float>::max();
for (const auto new_width : std::array<int, 2>{ceiled_width, floored_width})
{
  for (const auto new_height : std::array<int, 2>{ceiled_height, floored_height})
  {
    const float new_aspect_ratio = static_cast<float>(new_width) / new_height;
    const float distance = std::abs((new_aspect_ratio / aspect_ratio) - 1.f);
    if (distance < min_distance)
    {
       min_distance = distance;
       int_width = new_width;
       int_height = new_height;
     }
  }
  
  return {int_width, int_height};
```

Reply:
Yeah, my code was quick and stupid, I might try this.

Reply:
Done